### ISSUE TITLE
Scanmerge signature?

### ISSUE THREAD

**Issue by dmitriz** [uzujs/uzu#4]
It is flipped comparing to `scan`, stream first instead of last, a typo?

**Comment by jayrbolton** [comment]
Are you referring to:
`[[Stream(a), ((b, a) -> b]] -> b -> Stream(b)`

So it takes two arguments:
* An array of pairs of streams and updater functions (reducers)
* A default value

And it returns a single stream

For example, for arithmetic:
```
function arithmetic ({add, sub, mul, div}) => {
  return scanMerge([
    [add,    (sum, n) => sum + n]
  , [sub,    (sum, n) => sum - n]
  , [mul,    (sum, n) => sum * n]
  , [div,    (sum, n) => sum / n]
  ], 0)
}
```

This is pretty much a "reducer". But I don't want to call it "reducer" as I don't want it to be conflated with the `reduce` function. Maybe "aggregate" is a good name?

**Comment by dmitriz** [comment]
Yes, I felt confused about why writing data (stream) first, when in the regular `scan`, it is last, like you would normally do in FP.

But I've seen it was introduced that way in https://github.com/paldepind/flyd-scanmerge, which hasn't been updated for 2 yrs, there might have been some reasons for that argument flip back then, but I don't quite see it now, and my understanding is these days it is preferred
to put data last (as e.g. popularised by Dr. Frisby).

It does not seem so common among other stream libraries, and is not even included in the 100+ RxJS methods. I have only see it in https://github.com/MithrilJS/mithril.js/blob/next/stream/stream.js#L127,
where they have possibly just copied the old signature from flyd-scanmerge,
and might be open to flip it if asked, or explain the reasons not to if there are some.


To me it looks like a convenience derivative from both `scan` and `merge`,
dealing with the special case when all streams are of the same type and start with the same initial value, so basically yes -- a composition of the reducers for individual streams representing actions, 
that you can put together into a single reducer via some kind of general purpose function:

```js
const toReducer = pairs =>

    // returns reducer performing all the actions
    (state, action) => ...
```

You would need to label each stream's value with the action type,
and run each reducer depending on the type.

Yes, not a reducer, but can be represented as a scan over one
from the merged stream of labeled actions.

**Comment by dmitriz** [comment]
BTW, confusingly the stream `merge` is called `combine` in Hareactive:

https://github.com/funkia/hareactive#combinea-ba-streama-b-streamb-streamab

So now we have combine in the `xstream`, `mostjs`, `flyd` and `hareactive`, and all 4 are different. Great! ;(

**Comment by jayrbolton** [comment]
Aside: Another way to have "scanMerge" with using just "compose" and "lift" (here lift needs to be curried to 3 arguments). I think this would work the same, although i haven't tested it.

```js
function arithmetic ({add, sub, mul, div}) {
  return R.compose(
    lift(R.div, div)
  , lift(R.mul, mul)
  , lift(R.sub, sub)
  , lift(R.add, add)
  )(myStream)
}
```

**Comment by dmitriz** [comment]
That would only apply the lifted functions once to each new value of the input streams, as far as I understand, where with scan you want to iterate your reducer every time with values previously computed. Am I confused?

**Comment by jayrbolton** [comment]
Actually, here's a way to aggregate streams that may (or not) be cleaner than scanmerge. I think it reads pretty well

```js
function counter ({increment, decrement, reset}) {
  const updaters = merge([
    always(R.add(1), increment)
  , always(R.sub(-1), decrement)
  , always(R.multiply(0), reset)
  ])
  const count = scan((sum, fn) => fn(sum), 0, updaters)
  return { count }
}
```

This makes the input streams into a stream of updater functions. I'm thinking if there are other interesting uses of having streams of functions
